graphql/mutation_resolver.go

```diff
@@ -149,18 +149,29 @@ func (r *mutationResolver) SetAnnotationMetadataLinks(ctx context.Context, taskI
 
 // SaveAdminSettings is the resolver for the saveAdminSettings field.
 func (r *mutationResolver) SaveAdminSettings(ctx context.Context, adminSettings restModel.APIAdminSettings) (*restModel.APIAdminSettings, error) {
```

Comment:
basically, what I did was make this mutator similar to how the adminsettings legacy page works. there's 3 steps really:

- set evergreen settings in a "dry" mode, give us back settings object, with all fields populated
- set the evergreen settings, make it persistent and comapre it to the redacted secrets
- build updatedAdminSettings from service and return


this workflow is almost exactly as handled in the [legacy page here](https://github.com/evergreen-ci/evergreen/blob/fd03099865b8b61fd358415487e51d1c19a53a17/rest/route/admin_settings.go#L97-L108)

I think moving forward SetEvergreenSettings shouldn't log secrets we want redacted